pipeline/functional_assays/functional_assays_to_vcf.py

```diff
@@ -0,0 +1,128 @@
+                        help='Ouput VCF file result.')
+    parser.add_argument('-e', '--errors', type=argparse.FileType('w'),
+                        help='File containing all variants that could not be parsed.')
+    parser.add_argument('-g', '--gpath', default='/hive/groups/cgl/brca/phase1/data/resources/hg19.fa',
+                        help='Whole path to genome file. Default: (/hive/groups/cgl/brca/phase1/data/resources/hg19.fa)')
+    parser.add_argument('-r', '--rpath', default='/hive/groups/cgl/brca/phase1/data/resources/refseq_annotation.hg19.gp',
+                        help='Whole path to refSeq file. Default: (/hive/groups/cgl/brca/phase1/data/resources/refseq_annotation.hg19.gp)')
+    parser.add_argument('-s', '--source', default='FunctionalAssay')
+
+    options = parser.parse_args()
+    return options
+
+
+def main():
+    options = parse_args()
+    inputFile = options.input
+    annotFile_path = options.inAnnot
+    vcfFile = options.out
+    genome_path = options.gpath
+    refseq_path = options.rpath
+    errorsFile = options.errors
+    source = options.source
+
+    with open(refseq_path) as infile:
+        transcripts = hgvs_utils.read_transcripts(infile)
+
+    genome = SequenceFileDB(genome_path)
+
+    def get_transcript(name):
+        return transcripts.get(name)
+
+    # open and store annotation fields in a dictionary
+    annotDict = defaultdict()
+    with open(annotFile_path) as inAnnotFile:
+        for line in inAnnotFile:
+            line = line.strip().split('\t')
+            annotDict[line[0]] = line[1]
+
+    # print header lines to vcf file
+    print('##fileformat=VCFv4.0', file=vcfFile)
+    print('##source={0}'.format(source), file=vcfFile)
+    print('##reference=GRCh37', file=vcfFile)
+    for annotation, description in annotDict.items():
+        print('##INFO=<ID={0},Number=.,Type=String,Description="{1}">'.format(annotation.replace(' ', '_'), description), file=vcfFile)
+    print('#CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO', file=vcfFile)
+
+    # extract INFO field column indicies for annotation terms
+    headerline = inputFile.readline().strip().replace(' ', '_').replace('"', '').split('\t')
+
+    fieldIdxDict = defaultdict()
+    for index, field in enumerate(headerline):
+        fieldIdxDict[field] = index
+
+    # extract info from each line of the flat file
+    for line in inputFile:
+        line = line.replace('"', '')
+        INFO_field = list()
+        parsedLine = line.strip().split('\t')
+        for field in headerline:
+            field_index = fieldIdxDict[field]
+            field_value = parsedLine[field_index]
+            field_value = normalize(field, field_value)
+            INFO_field.append('{0}={1}'.format(field, field_value))
+
+        # extract hgvs cDNA term for variant and cleanup formatting
+        hgvsName = parsedLine[fieldIdxDict['hgvs_nucleotide']]
+        if hgvsName == '-':
+            print(parsedLine)
```

Comment:
Is there a `file=` argument missing? If not, maybe using `logging` to clearly separate diagnostics from output.

Reply:
makes sense, added explicit logging. FWIW this file is a tweaked copy of lovd2vcf.